components/ProductsCard.jsx

```diff
@@ -21,34 +19,25 @@ const PRODUCTS_QUERY = gql`
     }
   }
 `
-
 export function ProductsCard() {
```

Comment:
I wonder if we can simplify this component even further by removing a bunch of `useState` and `useEffect`.  E.g:

```
export function ProductsCard() {
  const [populateProduct, { isLoading: isAddingProducts }] = useShopifyMutation({
    query: PRODUCTS_QUERY,
  })

  const {
    data,
    refetch: refetchProductCount,
    isLoading: isLoadingCount,
  } = useAppQuery({
    url: '/api/products-count',
  })

  const toastMarkup = data?.count && (
    <Toast
      content="5 products created!"
    />
  )

  const handlePopulate = () => {
    Promise.all(
      Array.from({ length: 5 }).map(() =>
        populateProduct({
          input: {
            title: randomTitle(),
            variants: [{ price: randomPrice() }],
          },
        })
      )
    ).then(() => {
      refetchProductCount()
    })
  }

  return (
    <>
      {toastMarkup}
      <Card
        title="Product Counter"
        sectioned
        primaryFooterAction={{
          content: 'Populate 5 products',
          onAction: handlePopulate,
          loading: isAddingProducts,
        }}
      >
        <TextContainer spacing="loose">
          <p>
            Sample products are created with a default title and price. You can
            remove them at any time.
          </p>
          <Heading element="h4">
            TOTAL PRODUCTS
            <DisplayText size="medium">
              <TextStyle variation="strong">{isLoadingCount ? '-' : data.count }</TextStyle>
            </DisplayText>
          </Heading>
        </TextContainer>
      </Card>
    </>
  )
}
```